### ISSUE TITLE
PyYAML DeprecationWarning

### ISSUE THREAD

**Issue by mihaTrajbaric** [xlab-si/xopera-opera#176]
In PyYAML >= 5.4, importing from _yaml is deprecated and DeprecationWarning is raised:

DeprecationWarning: The _yaml extension module is now located at yaml._yaml and its location is subject to change.  To use the LibYAML-based parser and emitter, import from `yaml`: `from yaml import CLoader as Loader, CDumper as Dumper`.

A warning was encountered when importing opera (0.6.4) into xopera-rest-api (Python 3.8.7), I think [this line is to blame](https://github.com/xlab-si/xopera-opera/blob/5aa2c90d24512b2a245954bbd2e2354103a08265/src/opera/parser/yaml/loader.py#L5).

Stacktrace:
```
../../../.venv/lib/python3.8/site-packages/opera/commands/__init__.py:1: in <module>
    from opera.commands import (
../../../.venv/lib/python3.8/site-packages/opera/commands/deploy.py:10: in <module>
    from opera.commands.info import info
../../../.venv/lib/python3.8/site-packages/opera/commands/info.py:10: in <module>
    from opera.parser import tosca
../../../.venv/lib/python3.8/site-packages/opera/parser/tosca/__init__.py:4: in <module>
    from opera import stdlib
../../../.venv/lib/python3.8/site-packages/opera/stdlib/__init__.py:3: in <module>
    from opera.parser import yaml
../../../.venv/lib/python3.8/site-packages/opera/parser/yaml/__init__.py:1: in <module>
    from .loader import Loader
../../../.venv/lib/python3.8/site-packages/opera/parser/yaml/loader.py:6: in <module>
    from _yaml import CParser
../../../.venv/lib/python3.8/site-packages/_yaml/__init__.py:18: in <module>
DeprecationWarning: The _yaml extension module is now located at yaml._yaml and its location is subject to
 change. To use the LibYAML-based parser and emitter, import from `yaml`: `from yaml import CLoader as
 Loader, CDumper as Dumper`.
```

**Comment by anzoman** [comment]
@mihaTrajbaric thanks, good find, we have to fix this.

@sstanovnik now when xOpera has some static code testing - do you happen to know if we can somehow detect access to a protected member of a Python module too? This could help us supply better code without deprecations.

**Comment by sstanovnik** [comment]
I've created a fix at #180. It makes our code compatible with both `PyYAML<5.4` and `PyYAML>=5.4`, removing this warning. I don't think it's wise to restructure the parser to remove this internal dependency. It "has to be there for a reason", and I'm not inclined to dig into the reason at this point :)

We already have checks for accessing protected members, pylint does that, e.g. find `# pylint: disable=protected-access` in `definition_collector_mixin.py`. Why this wasn't caught is one of:

* pylint ignored it because it's in a try-catch block
* `_yaml` may not technically be a protected "member", because it's not `something._underscoredname`
